File: astroscope-lab/src/main/java/com/astroscope/lab/service/AdminExportService.java
package com.astroscope.lab.service;

import com.astroscope.lab.model.User;
import com.astroscope.lab.platform.reporting.ReportCriteriaAssembler;
import com.astroscope.lab.repository.LegacyExportRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class AdminExportService {

    private final LegacyExportRepository legacyExportRepository;
    private final ReportCriteriaAssembler reportCriteriaAssembler;

    public AdminExportService(LegacyExportRepository legacyExportRepository,
                            ReportCriteriaAssembler reportCriteriaAssembler) {
        this.legacyExportRepository = legacyExportRepository;
        this.reportCriteriaAssembler = reportCriteriaAssembler;
    }

    public List<User> exportUsers(String usernamePrefix) {
        String criteria = reportCriteriaAssembler.buildUserExportCriteria(usernamePrefix);
        return legacyExportRepository.findUsersForExport(criteria);
    }
}

File: astroscope-py/app/services/admin_export_service.py
from sqlalchemy import text

from app.extensions import db
from app.models import User
from app.platform.reporting.export_criteria import report_criteria_assembler


class AdminExportService:
    def export_users(self, username_prefix: str | None):
        criteria = report_criteria_assembler.build_user_export_criteria(username_prefix)
        sql = f"SELECT * FROM users WHERE 1=1 {criteria} ORDER BY username ASC"
        rows = db.session.execute(text(sql)).mappings().all()
        return [User.query.get(row["id"]) for row in rows if row]


admin_export_service = AdminExportService()
